index.js

```diff
@@ -1,107 +1,55 @@
         const repo = core.getInput("repo", { required: true });
         const owner = core.getInput("owner", { required: true });
         const pullNumber = core.getInput("pr_number", { required: true });
-
         const githubToken = core.getInput("token", { required: true });
-        const githubAPI = new GitHubAPI(githubToken);
-        const pullRequestData = await githubAPI.getPullRequest(owner, repo, pullNumber);
-        const filesContentGetter = (filePath) => githubAPI.getContent(owner, repo, filePath, pullRequestData.head.sha);
-        const inFileCommenter = (comment, filePath, line) => githubAPI.createReviewComment(owner, repo, pullNumber, pullRequestData.head.sha, comment, filePath, line);
         const openaiApiKey = core.getInput("openai_api_key", { required: true });
-        const openaiAPI = new OpenAIAPI(openaiApiKey, filesContentGetter, inFileCommenter, getApproxMaxSymbols());
 
-        const changedFiles = await githubAPI.listFiles(owner, repo, pullNumber);
-        const fileExtensions = core.getInput("file_extensions", { required: false });
+        const includeExtensions = core.getInput("include_extensions", { required: false });
+        const excludeExtensions = core.getInput("exclude_extensions", { required: false });
+        const includePaths = core.getInput("include_paths", { required: false });
         const excludePaths = core.getInput("exclude_paths", { required: false });
-        const filteredChangedFiles = getFilteredChangedFiles(
-            changedFiles,
-            fileExtensions,
-            excludePaths
-        );
-       
-        const allInOneSucess = await processAllInOneStrategy(filteredChangedFiles, openaiAPI, githubAPI, owner, repo, pullNumber);
-        if (allInOneSucess)
-            return;
 
-        const diffByDiffSucess = await processDiffByDiffStrategy(filteredChangedFiles, openaiAPI);
-        if (diffByDiffSucess)
-            return;
+        const githubAPI = new GitHubAPI(githubToken);
+        const changedFiles = await githubAPI.listFiles(owner, repo, pullNumber);
+        const filteredChangedFiles = getFilteredChangedFiles(changedFiles, includeExtensions, excludeExtensions, includePaths, excludePaths);
+
+        const pullRequestData = await githubAPI.getPullRequest(owner, repo, pullNumber);
+        const fileContentGetter = async (filePath) => await githubAPI.getContent(owner, repo, filePath, pullRequestData.head.sha);
+        const fileCommentator = (comment, filePath, line) => {
+            githubAPI.createReviewComment(owner, repo, pullNumber, pullRequestData.head.sha, comment, filePath, line);
+        }
+        const openAI = new OpenAIAgent(openaiApiKey, fileContentGetter, fileCommentator);
+        await openAI.doReview(filteredChangedFiles);
 
-        console.error("Have no strategy to process such a big PR.");
     } catch (error) {
-        console.error(error);
-        core.setFailed(error.message);
+        core.warning(error);
     }
 };
 
```

Comment:
Uses `await` with no asynchronous call. The attached handler, `fileContentGetter`, should probably be asynchronous, calling for a revision here to ensure it matches an async pattern if that's intended.